components/search/src/filter.rs

```diff
@@ -108,11 +110,91 @@ fn maybe_extract_engine_config(
         .map(|_variant| SearchEngineDefinition::from_configuration_details(&identifier, base))
 }
 
+fn determine_default_engines(
+    engines: &[SearchEngineDefinition],
+    default_engines_record: Option<JSONDefaultEnginesRecord>,
+    user_environment: &SearchUserEnvironment,
+) -> (Option<String>, Option<String>) {
+    match default_engines_record {
+        None => (None, None),
+        Some(record) => {
+            let mut default_engine_id = None;
+            let mut default_engine_private_id = None;
+
+            let specific_default = record
+                .specific_defaults
+                .into_iter()
+                .rev()
+                .find(|r| matches_user_environment(&r.environment, user_environment));
+
+            if let Some(specific_default) = specific_default {
+                // Check the engine is present in the list of engines before
+                // we return it as default.
+                let engine_id = find_engine_with_match(engines, specific_default.default);
+                if engine_id.is_some() {
+                    default_engine_id = engine_id;
+                }
+                let private_engine_id =
+                    find_engine_with_match(engines, specific_default.default_private);
+                if private_engine_id.is_some() {
+                    default_engine_private_id = private_engine_id;
+                }
```

Comment:
I noticed we're calling `is_some` to check if we've found a matching engine. Could we do the same as line `130` and use an `if let Some` here instead? I was curious and read: https://doc.rust-lang.org/stable/book/ch06-03-if-let.html maybe that could work and make the code more concise / clear? Thoughts? 
```
if let Some(engine_id) = find_engine_with_match(engines, specific_default.default) {
    default_engine_id = engine_id;
}
if let Some(private_engine_id) = find_engine_with_match(engines, specific_default.default_private) {
    default_engine_private_id = private_engine_id;
}

```

Reply:
`default_engine_id` and `default_engine_private_id` are options, so it would be a little wordier to assign to them directly:

```rust
if let Some(engine_id) = find_engine_with_match(engines, specific_default.default) {
    default_engine_id = Some(engine_id);
}
```

But [`Option::replace`](https://doc.rust-lang.org/std/option/enum.Option.html#method.insert) can help!

```rust
if let Some(engine_id) = find_engine_with_match(engines, specific_default.default) {
    default_engine_id.replace(engine_id);
}
```